rundeckapp/grails-app/services/rundeck/services/PluginApiService.groovy

```diff
@@ -268,7 +272,23 @@ class PluginApiService {
                  rundeckCompatibilityVersion: rdVer,
                  targetHostCompatibility: tgtHost,
                  pluginDate   : toEpoch(dte),
-                 enabled      : true]
+                 enabled      : true] as LinkedHashMap<Object, Object>
+
+                if(service != "UI") {
```

Comment:
need to do an import but preferred to using constant string:

```suggestion
                if(service != ServiceNameConstants.UI) {
```

but, is there a reason to skip UI plugins here?

Reply:
That's a very good question. I brought this bit of logic over from the non-public endpoint that was being used. IIRC I got an error without the check, but in either case I don't enjoy this bit as it's not clear at all why "UI" is an exception. Would need to do further debugging to figure out why.